### PR TITLE
chore(docs) change example to use eventListener instead of polling

### LINKED ISSUE #10312: Issue with "Document.hasFocus()": `setInterval(checkPageFocus, 300)` is not a very efficient method.
MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/Document/hasFocus

#### What information was incorrect, unhelpful, or incomplete?
I came to the page to figure out if there was a relation between `document.hasFocus()` and the `focus` and `blur` events.
To my surprise I saw the example containing `setInterval(checkPageFocus, 300);` which seems like an inefficient way to do it.

#### Specific section or headline?
https://developer.mozilla.org/en-US/docs/Web/API/Document/hasFocus#javascript

#### What did you expect to see?
Something more along the lines of
```js
window.addEventListener("blur", checkPageFocus);
window.addEventListener("focus", checkPageFocus);
```

#### Did you test this? If so, how?
I have implemented in this way in my own application. However, I'm not sure if a guarantee is being made regarding the spec that this behaviour is expected. I.e. I'm not sure if `document.hasFocus()` is guaranteed to be `true` right after a `focus` event and `false` right after a `blur` event. [This page](https://developers.google.com/web/updates/2018/07/page-lifecycle-api#state-passive) seems to suggest so but I couldn't find anything in the spec.

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/api/document/hasfocus`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/Document/hasFocus
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/document/hasfocus/index.md
* Last commit: https://github.com/mdn/content/commit/2279e5ae6c229c707a014a22aa1ec4635a0f981f
* Document last modified: 2021-09-14T18:08:09.000Z

</details>

### PR DESCRIPTION
**Pull request by bsmth**
The existing example uses polling every 300 millis, it's noted in the linked issue that we can use eventListeners instead.

__Notes:__
This example could be improved in general as we're using an iframe, so we might not even need to open a new window. We have the demonstrated effect by clicking inside the iframe and the embedding document.

Fixes #10312

### CODE REVIEW

**Review by wbamberg — COMMENTED** [review]
I don't know what I think about this. I agree that polling is bad, but in this version we check whether we have focus in `blur` and`focus` handlers - isn't that pointless, since we know what the answer is going to be?

I'm not sure what is a better option though. We could check on a single timer, but that makes the interaction a little more complicated (i.e. we'd have to say: "push the button to start the timer, then click outside the iframe"). But it still might be more sensible.

File: files/en-us/web/api/document/hasfocus/index.md [hunk 1]
```diff
@@ -39,9 +39,10 @@ None.
...
 
-This example checks whether the document has focus every 300 milliseconds. To test the
-functionality of `hasFocus()`, click on the button to open a new window, and
-try switching between the two pages.
+The following example checks whether the current document has focus.
+We create a function called `checkPageFocus()` that updates an element depending on the result of `document.hasFocus()`.
+Two listeners are triggered on [`focus`](/en-US/docs/Web/API/Element/focus_event) and [`blur`](/en-US/docs/Web/API/Element/blur_event) events that call our `checkPageFocus()` function.
+To test `hasFocus()`, click on the button to open a new window and try switching between both pages.
 
 ### HTML
```
Comment (wbamberg) [inline files/en-us/web/api/document/hasfocus/index.md:47]: You didn't add this, but if we are rewriting the example let's avoid the inline event handler at:

```html
<button onclick="openWindow()">Open a new window</button>
```

**Comment by bsmth** [comment]
> isn't that pointless, since we know what the answer is going to be?

Very true. I'll have a think about it and circle back to this again.

File: files/en-us/web/api/document/hasfocus/index.md (hunk 1 above)
Comment (bsmth) [reply files/en-us/web/api/document/hasfocus/index.md:43]: Done in 1897d446504150cb0fb1f2a42433e5a981486132

**Comment by bsmth** [comment]
Hi @wbamberg I've taken another shot at it. Polling makes more sense than an `eventListener`, so I've reverted to the original here unless there's a better way of doing it I'm not thinking of.

**Comment by wbamberg** [comment]
> Hi @wbamberg I've taken another shot at it. Polling makes more sense than an `eventListener`, so I've reverted to the original here unless there's a better way of doing it I'm not thinking of.

The only alternative I can think of is to have an additional button saying "Check in 2 seconds". So rather than polling all the time, you push the button and then push "Open new window", then it checks.

But I think this is probably worse!

- it's a more complicated example to understand
- the user interaction is more complicated
- you have to give the window focus when you click the button, which makes it weird

**Comment by bsmth** [comment]
> * user interaction is more complicated

+1 on that. Out of the variations I was trying, the current one is the most self-explanatory, I think. I'm fine with the changes in this PR for now.

**Review by wbamberg — APPROVED** [review]
👍 thank you @bsmth ! I agree this is the best either of us can think of :).

**Pull request merged**